--- create-pwiz-localdecls	2003-12-26 08:30:42.000000000 +0100
+++ create-pwiz-localdecls	2004-07-07 13:25:48.652336829 +0200
@@ -35,7 +35,7 @@
 ###
 
 echo -n 'Checking libc version... '
-gcc -Wall -o libctest libctest.c
+gcc -Wall -o libctest $1/libctest.c
 VERSION=`ldd libctest | grep libc\\.so | awk '{print $1}'`
 rm libctest
 echo $VERSION
--- pwizwatch.c	2004-05-18 16:43:08.000000000 +0200
+++ pwizwatch.c	2004-06-10 15:41:54.000000000 +0200
@@ -23,6 +23,9 @@
  *
  * november-25-2002 - Modifications by Olivier Fleurigeon
  *                                  <olivier.fleurigeon@cegedim.fr>
+ *
+ * May-18-2004 Adapted for read support for Packaging Wizard by
+ * Stanislav Brabec <sbrabec@suse.cz>
  */
 
 #include <sys/param.h>
@@ -2603,10 +2606,8 @@
 	instw_print(&instw);
 #endif
 
-	if(flags & (O_WRONLY | O_RDWR)) {
-		backup(instw.truepath);
-		instw_apply(&instw);
-	}
+	backup(instw.truepath);
+	instw_apply(&instw);
 
 	instw_getstatus(&instw,&status);
 
@@ -2617,6 +2618,8 @@
 	
 	if(flags & (O_WRONLY | O_RDWR)) 
 		log("%d\topen\t%s\t#%s\n",result,instw.reslvpath,error(result));
+	else
+		log("%d\tread\t%s\t#%s\n",result,instw.reslvpath,error(result));
 
 	instw_delete(&instw);
 
